### PR TITLE
kv: Clean up kv.go

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: kv/kv.go [hunk 1]
```diff
@@ -13,98 +13,12 @@
...
-// DecodeFn is a function that decodes data after fetching from store.
-type DecodeFn func(raw interface{}) (interface{}, error)
-
-// EncodeFn is a function that encodes data before putting into store.
-type EncodeFn func(raw interface{}) (interface{}, error)
+import "github.com/juju/errors"
 
 // ErrNotCommitted is the error returned by CommitVersion when this
 // transaction is not committed.
 var ErrNotCommitted = errors.New("this transaction is not committed")
```
Comment (disksing) [inline kv/kv.go]: I've noticed that all other errors are listed in 'iter.go'. (not a good place either)
I think we should find a better place to declare all errors.

Comment (shenli) [inline kv/kv.go]: terror.go is a better place.

**Comment by ngaut** [comment]
PTAL

**Comment by disksing** [comment]
`IsErrNotFound()` in union_store.go better be moved to error.go too.

**Comment by coocood** [comment]
We should change those errors to `*terror.Error` type later if not in this PR.

**Comment by ngaut** [comment]
Yes, later.

**Comment by disksing** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**